From dc53cdaa9f2e23ba749e4d684a0177608df1b74d Mon Sep 17 00:00:00 2001
From: Manjunath <manjunathx.b.s@intel.com>
Date: Tue, 21 May 2013 09:57:52 +0530
Subject: When trying to play corrupted ogg file, the media player is crashing or not responding.

BZ: 109320

The function _vorbis_unpack_info is returning BAD_HEADER error for the corrupted
file because of which CHECK_EQ is giving Fatal Signal. This patch handles this
error by returning ERROR_MALFORMED.

Category: aosp improvement
Domain: AUDIO_MEDIA-Audio Framework
Origin: Internal
Upstream-Candidate: yes

Change-Id: Ic41b42ff965da8f686231abe4c7ec2297dd295a2
Orig-Change-Id: I2c5fefbfd89f0a2412f9a9d57412fcc2a262d478
Signed-off-by: Manjunath <manjunathx.b.s@intel.com>
---
 media/libstagefright/OggExtractor.cpp |    7 ++++++-
 1 files changed, 6 insertions(+), 1 deletions(-)

diff --git a/media/libstagefright/OggExtractor.cpp b/media/libstagefright/OggExtractor.cpp
index 8ddf120..422fc92 100644
--- a/media/libstagefright/OggExtractor.cpp
+++ b/media/libstagefright/OggExtractor.cpp
@@ -731,7 +731,12 @@ status_t MyVorbisExtractor::verifyHeader(
     switch (type) {
         case 1:
         {
-            CHECK_EQ(0, _vorbis_unpack_info(&mVi, &bits));
+            int retval  = _vorbis_unpack_info(&mVi, &bits);
+            CHECK_NE(OV_EFAULT, retval);
+
+            if (retval == OV_EBADHEADER || retval == OV_EVERSION) {
+                return ERROR_MALFORMED;
+            }
 
             mMeta->setData(kKeyVorbisInfo, 0, data, size);
             mMeta->setInt32(kKeySampleRate, mVi.rate);
-- 
1.7.4.1

